### DISCUSSION TITLE
Welcome to brevitas-radioml-challenge-21 Discussions!

### DISCUSSION THREAD

**Discussion by maltanar** [Xilinx/brevitas-radioml-challenge-21#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by louisinhit** [comment]
Hi dear organizers, 

So the official will not provide any remote server or GPU or something similar? All the training and testing use our own local host, right?

Thanks.

**Reply by volcacius** [reply]
Hello,
I opened a discussion on the topic here https://github.com/Xilinx/brevitas-radioml-challenge-21/discussions/4 .

**Comment by ichwana** [comment]
Hi dear organizers,

Is there any guide to do it on the MacBook platform?

Thanks

**Reply by maltanar** [reply]
Hi @ichwana , the infrastructure we provide is Docker-based which should in theory also work on MacBooks, but this is not something we have tested. Please also note that training on a laptop may take quite long, we normally recommend doing the training experiments on a server/cloud with access to GPUs.

**Comment by ENGRsao** [comment]
I keep getting error like this , anytime i run the run-docker.sh file to create the image .

docker: Error response from daemon: OCI runtime create failed: invalid mount {Destination:\Program Files\Git\workspace\sandbox Type:bind Source:/run/desktop/mnt/host/c/brevitas-radioml-challenge-21/sandbox;C Options:[rbind rprivate]}: mount destination \Program Files\Git\workspace\sandbox not absolute: unknown.

I will be glad if anyone can guide me on the steps to take to successfully mount the dataset directory. thanks . I am running this on window 4gb RAM

**Comment by ashok-kummar** [comment]
Hello everyone,
    I'm getting this error when I run /run_docker.sh. Is there any mistake in my folder structure?

_Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
Docker container is named sandbox_vscode
Port-forwarding for Jupyter 8888:8888
Port-forwarding for Netron 8081:8081
Mounting /com.docker.devenvironments.code/sandbox at /workspace/sandbox
Mounting /workspace/dataset at /workspace/dataset
Exposing GPU(s): device=0
**[C 07:35:34.372 NotebookApp] No such file or directory: /workspace/sandbox/notebooks**_

The folder structure inside the docker container is shown below. I have created the workspace folder and its sub-folders.

![image](https://user-images.githubusercontent.com/59319993/130583325-84c5459f-d032-457f-9df0-0e8b12251d93.png)

**Comment by maltanar** [comment]
Hello @ashok-kummar, which operating system are you running this under? How are you cloning the repository and setting up the environment variables? 

```Mounting /com.docker.devenvironments.code/sandbox at /workspace/sandbox``` <- is `/com.docker.devenvironments.code` a valid path?

Note that we've only tested the sandbox in Linux-based environments. If you are using a special VSCode plugin to build/launch the container this may also be the cause, the intention is to simply run the `run-docker.sh` script from a terminal. There should be no need to create the `workspace` folder, that comes with the Docker base image.

**Reply by ashok-kummar** [reply]
Hi @maltanar , I'm using Docker Desktop application (Windows 11 OS). I'm creating a Dev environment as shown below.

![image](https://user-images.githubusercontent.com/59319993/130794984-bc2989fb-10e1-450c-83fb-ebfa506d46bd.png)

Once the git repo is cloned, the Docker app opens VS code and I ran `run-docker.sh` from the VS Code terminal.

`/com.docker.devenvironments.code/sandbox` -> This is a valid path inside the container.

**Reply by maltanar** [reply]
I see. This may be tricky to debug as I'm not sure what Docker Desktop and VSCode are doing under the hood. If you have WSL2 enabled, my recommendation would be to install Ubuntu 18.04 or 20.04 from the Windows Marketplace and directly clone the repo & run the script from the WSL Linux command line. I was able to get the baseline notebook running this way in Windows 10.

And let's not hijack this discussion post further. If you have further questions about this, please open an issue here on GitHub instead.
